shopelectro/selenium/elements/product.py

```diff
@@ -26,17 +31,43 @@ def remove_from_cart(self):
 
 class CatalogCard(Product):
 
-    def __init__(self, driver: SiteDriver, card_index: int):
+    def __init__(
+        self,
+        driver: SiteDriver,
+        *,
+        index: int = None,
+        id_: int = None,
+    ):
         """
         Ctor.
 
-        :param int card_index: The index number of the product card at a category page
+        :param int index: The index number of the product card at a category page
         """
         self.driver = driver
-        self.xpath = f'//*[@id="products-wrapper"]/div[{card_index}]/div[2]/div[5]/'
+
+        if (index is None and not id_) or (index and id_):
```

Comment:
is semantic coupling between arguments. It would be more good to create two separated constructors: one for `index` and another for `id_`.

Python doesn't allow us to create multiple constructors, but we can use static class methods instead:

```
class CatalogCard(Product):

    def __init__(self, index: int, *args, **kwargs):
        ...

    @classmethod
    def with_id(cls, id_, *args, **kwargs):
        return cls(...)
```